react_juce/core/ImageView.cpp

```diff
@@ -34,47 +34,58 @@ namespace reactjuce
         if (name == sourceProp)
         {
             const juce::String source = value.toString();
-            const juce::URL    sourceURL = source;
+            const int sourceHash      = juce::DefaultHashFunctions::generateHash(source, 1024);
+            const juce::URL sourceURL = source;
 
-            if (isWellFormedURL(sourceURL))
+            // No need to recalculate the image if it's already in the ImageCache.
+            const juce::Image& cachedImage = juce::ImageCache::getFromHashCode(sourceHash);
+            if (cachedImage.isValid())
```

Comment:
Yep, perfect.

One comment here which is mostly a style comment: in this case if we can find the image in the cache, we don't even need to run the rest of the function. In these cases, I prefer an early return:

```
if (cachedImage.isValid())
    return setDrawableImage(cachedImage, sourceHash);

// No longer need the "else" block because control flow will only reach down here if
// we need to run the rest of the function
```

Basically let's us keep the indenting easier to read